src/types.cpp

```diff
@@ -362,7 +363,8 @@ SizedType CreateRecord(const std::string &name, std::weak_ptr<Struct> record)
 
 SizedType CreateStack(bool kernel, StackType stack)
 {
-  auto st = SizedType(kernel ? Type::kstack : Type::ustack, kernel ? 8 : 16);
+  // These sizes are based on the stack key (see CodegenLLVM::kstack_ustack)
+  auto st = SizedType(kernel ? Type::kstack : Type::ustack, kernel ? 12 : 20);
```

Comment:
Out of scope: I'm wondering if we could define the structs somewhere and then just use `sizeof(...)` to get the correct size (and possibly offsets, too). This applies to all kinds of codegen-defined structs so it's probably worth a separate larger refactoring.

Reply:
Yeah agreed. Having to change these in multiple places is not great.